imports/lib/schemas/withCommon.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from 'zod';
+import type { MongoRecordZodType } from './generateJsonSchema';
+import { TimestampFields } from './withTimestamps';
+import { UserFields } from './withUsers';
+
+// We could compose withTimestamps and withUsers, but in the case that the
+// passed in schema is not a ZodObject, we'd end up creating multiple nested
```

Comment:
Ahhh, and `Setting` in particular, with a discriminated union at the root, is not a `ZodObject`